File: week2/day7.js
const nameInput = document.getElementById("nameInput");
const weightInput = document.getElementById("weightInput");
const repsInput = document.getElementById("repsInput");
const calcBtn = document.getElementById("calcBtn");
const result = document.getElementById("result");

calcBtn.addEventListener("click", async function () {
  const name = nameInput.value;
  const weight = Number(weightInput.value);
  const reps = Number(repsInput.value);

  // 最低限の入力チェック（Day7の“完成感”が出る）
  if (!name || weight <= 0 || reps <= 0) {
    result.textContent = "名前・重さ・回数を正しく入力してね！";
    return;
  }

  const response = await fetch("http://127.0.0.1:8000/calc", {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ name: name, weight: weight, reps: reps }),
  });

  const data = await response.json();
  result.textContent = data.message;
});

File: week2/day7.py
from fastapi import FastAPI
from pydantic import BaseModel
from fastapi.middleware.cors import CORSMiddleware

app = FastAPI()

app.add_middleware(
    CORSMiddleware,
    allow_origins=["*"],
    allow_methods=["*"],
    allow_headers=["*"],
)

class WorkoutData(BaseModel):
    name: str
    weight: float
    reps: int

@app.post("/calc")
def calc_point(data: WorkoutData):
    point = data.weight * data.reps * 0.1
    return {
        "message": f"{data.name}さんのポイントは {point:.1f} です！",
        "point": point
    }
